### ISSUE TITLE
Undocumented forced non-root user & user groups

### ISSUE THREAD

**Issue by actuallymentor** [rocket-pool/smartnode-install#13]
Hello team!

I just tried to upgrade my node and received a `rocketpool should not be run as root. Please try again without 'sudo'`.

1. This is undocumented in the public beta guide
2. For those of us who are in the beta, this is very sudden 'oh shit everything broke with the update' moment
3. On discord someone mentioned that Rocketpool assumes the running user is part of the `docker` user group, is this correct and if so can we document it in the setup guide please?

**Comment by moles1** [comment]
@actuallymentor We implemented this change because some people were (incorrectly) running `rocketpool` as root and making a mess of their filesystem by installing RP scaffolding to the wrong path etc.

> This is undocumented in the public beta guide

Yes, although I feel that the warning is pretty self-explanatory? Is there a reason that we need to tell node operators ahead of time that they can't run the CLI as root? If they try it they will just run into this roadblock and then stop doing it.

> For those of us who are in the beta, this is very sudden 'oh shit everything broke with the update' moment

I'm not sure I understand this comment. Is there a reason that this change would break anything? imo nobody should be running the CLI as root (is there a use case where you would want/need to?). If anyone is, it should be easy enough for them to just switch to their own user account.

> On discord someone mentioned that Rocketpool assumes the running user is part of the docker user group, is this correct and if so can we document it in the setup guide please?

The install script runs `sudo usermod -aG docker $USER` so everyone should be in the group already if they ran the installer. If they run it with `-d` and follow the official docker install instructions, it instructs them to do the same thing.

**Comment by actuallymentor** [comment]
> We implemented this change because some people were (incorrectly) running rocketpool as root and making a mess of their filesystem by installing RP scaffolding to the wrong path etc.

A good reason, I'm not against it, just against a breaking change (if you installed as root) not being documented.

> Yes, although I feel that the warning is pretty self-explanatory? Is there a reason that we need to tell node operators ahead of time that they can't run the CLI as root? If they try it they will just run into this roadblock and then stop doing it.

I lazily set up as root on a VPS, ran the update script yesterday and the while CLI refused to play ball while I'm running 30 validators.

> I'm not sure I understand this comment. Is there a reason that this change would break anything? imo nobody should be running the CLI as root (is there a use case where you would want/need to?). If anyone is, it should be easy enough for them to just switch to their own user account.

VPS, root user only, run update script = CLI stops working. I didn't dive under the hood of the architecture so only know RP is "something with a Go binary and something with a few Docker containers". Switching a root install to a user install is not "easy enough" if you want to be sure you have minimal validator downtime...

> The install script runs sudo usermod -aG docker $USER so everyone should be in the group already if they ran the installer. If they run it with -d and follow the official docker install instructions, it instructs them to do the same thing.

But what about an existing install that runs as root as I (lazily, irresponsibly) did? I got on board with the beta as soon as I could, and only updated because someone on the discord reminded me there were rolling updates.

**Comment by actuallymentor** [comment]
Recommendations for the `known-issues` discord:

- Mention the medalla finality freeze
- Mention breaking changes in the binary (like sudden sudo requirement)
  - Mention "if you used version x and are upgrading root is now mandatory"
  - Perhaps a pointer on how to switch the running user of the RP system

**Comment by moles1** [comment]
> A good reason, I'm not against it, just against a breaking change (if you installed as root) not being documented

Fair point. FWIW, this change is documented in the release notes (https://github.com/rocket-pool/smartnode-install/releases/tag/0.0.4), just not in the announcement on the discord. I just listed the most important changes there and linked to the full notes for anyone who wants to see *all* the changes.

I've raised an issue on the smartnode repo to add in a flag for the next release which overrides the check and allows the CLI to be run as root; this should resolve any similar issues anyone has: https://github.com/rocket-pool/smartnode/issues/68
